sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/continuous/EpochCoordinator.scala

```diff
@@ -45,6 +45,11 @@ private[sql] case object IncrementAndGetEpoch extends EpochCoordinatorMessage
  */
 private[sql] case object StopContinuousExecutionWrites extends EpochCoordinatorMessage
 
+/**
+ * Returns boolean indicating if size of the epochs queue has exceeded maximum epoch backlog.
+ */
+private[sql] case object CheckIfMaxBacklogIsExceeded extends EpochCoordinatorMessage
```

Comment:
I'm not sure we need to make a side-channel in the RPC handler for this. I'd try to just make the query fail when the condition is reached in the first place.

Reply:
I think we'd probably want to add some method like private[streaming] stopWithException(e) to ContinuousExecution.

Reply:
Okay, thought about something like this but wasn't sure if it's fine to do so for the sake of this change. Thanks